cpu/msp430-common/lpm_cpu.c

```diff
@@ -47,16 +48,9 @@ enum lpm_mode lpm_set(enum lpm_mode target)
 
     /* ensure that interrupts are enabled before going to sleep,
        or we're bound to hang our MCU! */
-    if (target != LPM_ON) {
-        if ((__read_status_register() & GIE) == 0) {
-            printf("WARNING: entering low-power mode with interrupts disabled!\n");
-            printf("         Forcing GIE bit to 1!\n\n");
-            __bis_status_register(GIE);
-        }
-    }
+    assert((target == LPM_ON) || (__read_status_register() & GIE));
```

Comment:
This seems to be incorrect? The original statement says 
```c
if ((target != LPM_ON) && (GIE not set)) { -> error; set GIE;}
```
you say
```c
error if ((target != LPM_ON) || (GIE not set))
```
thats a different thing...